File: Array and String/Implement_strStr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if len(needle) == 0:
            return 0
        else:
            l =len(needle)
            for i in range(len(haystack)):
                if needle[0] == haystack[i]:
                    if needle == haystack[i:i+l]:
                        return i
            return -1

File: Array and String/Implement_strStr.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int haystack_size = haystack.size(),needle_size = needle.size();
        if(needle_size == 0){
            return 0;
        }
        for(int i=0;i<haystack_size-needle_size+1;i++){
            if(haystack.compare(i,needle_size,needle) == 0){
                return i;
            }
        }
        return -1;
    }
};